tests/memoryview/memoryview_annotation_typing.py

```diff
@@ -37,6 +42,39 @@ def two_dim(a: cython.double[:,:]):
     a[0,0] *= 3
     return a[0,0], a[0,1], a.ndim
 
+def slice_none(m: cython.double[:]):
+    """
+    >>> try:
+    ...     a = slice_none(None)
+    ... except TypeError as exc:
+    ...     assert COMPILED
+    ...     if "Argument 'm' must not be None" not in str(exc): raise
+    ... else:
+    ...     assert a == 1
+    ...     assert not COMPILED
+    """
+    return 1 if m is None else 2
+
+
+def slice_optional(m: typing.Optional[cython.double[:]]):
+    """
+    >>> slice_optional(None)
+    1
+    >>> a = numpy.ones((10,), numpy.double)
+    >>> slice_optional(a)
+    2
+
+    # Make sure that we actually evaluate the type and don't just accept everything.
+    >>> try:
+    ...     x = slice_optional(123)
+    ... except TypeError as exc:
+    ...     if "a bytes-like object is required, not 'int'" not in str(exc): raise
+    ...     assert COMPILED
```

Comment:
The exception message is different in Py2. Let's keep it simple. We don't need to check the exact message, just need to be sure that an invalid value isn't accepted. We already verify valid values above, so it "works in general".
```suggestion
    ...     if not COMPILED: raise
```